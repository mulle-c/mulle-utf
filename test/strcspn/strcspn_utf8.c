#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <string.h>


static void   test( char *s, char *p)
{
   size_t  result;

   printf( "mulle_utf8_strcspn( ");
   if( ! s)
      printf( "NULL");
   else
      printf( "\"%s\"",  s);

   if( ! p)
      printf( ", NULL");
   else
      printf( ", \"%s\"", p);

   printf( ") = ");

   result = mulle_utf8_strcspn( s, p);
   printf( "%ld (%ld)\n", (long) result, (long) ((s && p) ? strcspn( s, p) : 0));
}



int   main( int argc, char **argv)
{
   test( NULL, NULL);
   test( "", NULL);
   test( NULL, "");
   test( "", "");
   test( "", "");

   test( NULL, "");

   test( "abcd", "a");
   test( "abcd", "b");
   test( "abcd", "d");

   test( "abcd", "xyz");
   test( "abcd", "xcz");

   test( "abcd", "ab");
   test( "abcd", "bc");

   test( "abcd", "abc");

   test( "abc", "b");
   test( "abc", "abc");

   return( 0);
}