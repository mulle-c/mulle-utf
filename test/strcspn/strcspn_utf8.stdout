mulle_utf8_strcspn( NULL, NULL) = 0 (0)
mulle_utf8_strcspn( "", NULL) = 0 (0)
mulle_utf8_strcspn( NULL, "") = 0 (0)
mulle_utf8_strcspn( "", "") = 0 (0)
mulle_utf8_strcspn( "", "") = 0 (0)
mulle_utf8_strcspn( NULL, "") = 0 (0)
mulle_utf8_strcspn( "abcd", "a") = 0 (0)
mulle_utf8_strcspn( "abcd", "b") = 1 (1)
mulle_utf8_strcspn( "abcd", "d") = 3 (3)
mulle_utf8_strcspn( "abcd", "xyz") = 4 (4)
mulle_utf8_strcspn( "abcd", "xcz") = 2 (2)
mulle_utf8_strcspn( "abcd", "ab") = 0 (0)
mulle_utf8_strcspn( "abcd", "bc") = 1 (1)
mulle_utf8_strcspn( "abcd", "abc") = 0 (0)
mulle_utf8_strcspn( "abc", "b") = 1 (1)
mulle_utf8_strcspn( "abc", "abc") = 0 (0)
